charts/kyverno-policies/values.yaml

```diff
@@ -114,3 +114,7 @@ skipBackgroundRequests: ~
 # -- Kyverno version
 # The default of "autodetect" will try to determine the currently installed version from the deployment
 kyvernoVersion: autodetect
+
+# -- Kubernetes version
+# Against which the policy should work. Ideally range of versions no more than two prior (ex., 1.28-1.31), must be enclosed in quotes
+kubernetesVersion: "1.28-1.31"
```

Comment:
Why do we want a Helm option to configure it? It should be set as part of the release process, we can add a `testedKubernetesVersions: "1.28-1.31"` attribute in `charts/kyverno-policies/Chart.yaml`.

Reply:
Helm options are convenient way to allow users customize behavior of deployments.

Since this value is propagated to manifests and effectively impact behavior of deployment I thought it was a good idea to let users customize this field if they are running older or newer versions of Kubernetes without need to modify Chart files.

Otherwise, like it happened in our organization for example, we deployed policies v3.2.5 to newer version of Kubernetes 1.28 and found out they are not effective, since manifests has hardcoded versions 1.22-1.23. And there is no way to modify this without pulling and changing chart.

Reply:
Okay, maybe I misunderstood.

I just have checked latest version of Kyverno Policies chart 3.3.0 and I still see it has annotation kyverno.io/kubernetes-version: "1.22-1.23". Effectively this means even though I applied policies to my clusters they are not working.

Current K8s versions are 1.29-1.31 and even 1.28 is end of life already.

I have to change hardcoded templates myself to make this annotation match versions of my clusters.

I thought it would be good idea to make it possible for users customize while still keep default value match the officially tested Kubernernetes versions.

Release team will still need record this value in default values.yaml instead of tens of templates.

Reply:
I see those are two different configurations. One is mandatory for the release team to update the officially tested Kubernetes versions, and another is an optional configuration offered to users to customize this value.

Reply:
Totally makes scenes. I have updated Pull Request. Release team supposed to ensure values in Chart.yaml are updated and templates use them by default. Optional parameter added to Values in case user want to customize it. Please review.